app/src/org/commcare/android/tasks/TemplatePrinterTask.java

```diff
@@ -0,0 +1,239 @@
+package org.commcare.android.tasks;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.commcare.android.util.TemplatePrinterUtils;
+
+
+/**
+ * Asynchronous task for populating an html document with data.
+ *
+ * @author Richard Lu
+ * @author amstone
+ */
+public class TemplatePrinterTask extends AsyncTask<Void, Void, Boolean> {
+
+    /**
+     * If an error occurs, stores which type the error was, to be passed back to
+     * TemplatePrinterActivity through the PopulateListener's onError method
+     */
+    private int errorType;
+
+    /**
+     * The 2 types of errors that can be encountered in this task
+     */
+    public static final int IO_ERROR = 0;
+    private static final int VALIDATION_ERROR = 1;
+
+    /**
+     * The template file for this print action
+     */
+    private final File inputFile;
+
+    /**
+     * The path where the populated template should be saved to
+     */
+    private final String outputPath;
+
+    /**
+     * The mapping from keywords to case property values to be used in populating the template
+     */
+    private final Bundle values;
+
+    private final PopulateListener listener;
+
+
+    public TemplatePrinterTask(File input, String outputPath, Bundle values,
+                               PopulateListener listener) {
+        this.inputFile = input;
+        this.outputPath = outputPath;
+        this.values = values;
+        this.listener = listener;
+    }
+
+    /**
+     * Attempts to perform population of the template file, and throws the appropriate exception
+     * if encountering an error.
+     */
+    @Override
+    protected Boolean doInBackground(Void... params) {
+        try {
+            populateHtml(inputFile, values);
+            return true;
+        } catch (IOException e) {
+            errorType = IO_ERROR;
```

Comment:
seems odd having this errorType state-ness... maybe this should just return an integer with EG 0 being success, 1 being IO error, etc.?